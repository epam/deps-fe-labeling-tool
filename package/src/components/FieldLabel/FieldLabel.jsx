import React from 'react'
import PropTypes from 'prop-types'
import { Placement } from '@/enums/Placement'
import { Label, StyledTooltip } from './FieldLabel.styles'

const RESOURCE_REQUIRED_FIELD = 'Required Field'

const renderLabelContent = (name, required) => {
  if (required) {
    return (
      <StyledTooltip
        placement={Placement.LEFT}
        title={RESOURCE_REQUIRED_FIELD}
        required={required}
      >
        {name}
      </StyledTooltip>
    )
  }

  return (
    <span>{name}</span>
  )
}

const FieldLabel = ({
  active,
  clickable,
  name,
  required,
  className
}) => (
  <Label
    active={active}
    clickable={clickable}
    className={className}
  >
    {renderLabelContent(name, required)}
  </Label>
)

FieldLabel.propTypes = {
  active: PropTypes.bool,
  clickable: PropTypes.bool,
  name: PropTypes.oneOfType([
    PropTypes.string, PropTypes.element
  ]),
  required: PropTypes.bool,
  className: PropTypes.string
}

export { FieldLabel }
