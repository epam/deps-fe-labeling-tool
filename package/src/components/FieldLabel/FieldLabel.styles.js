import styled, { css } from 'styled-components'
import { Tooltip } from '@/components/Tooltip'
import { COLORS } from '@/theme/theme.default'

const baseActiveLabelStyles = css`
  font-weight: bold;
  text-decoration: underline;
`

export const Label = styled.div`
  width: 100%;
  display: flex;
  justify-content: space-between;
  padding-bottom: 0.4rem;

  & > span {
    cursor: default;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    font-size: 1.2rem;
    line-height: 2.5rem;
    font-weight: 600;
    padding: 0;
    color: rgb(38, 45, 54);
    text-transform: uppercase;

    & > span {
      font-size: 1.8rem;
    }

    ${(props) => props.active && baseActiveLabelStyles}
  }

  ${(props) => props.clickable && css`
    & > span:hover {
      cursor: pointer;
      ${baseActiveLabelStyles}
    }
  `}
`

export const StyledTooltip = styled(Tooltip)`
  ${({ required }) => required && css`
      &::before {
        content: '*';
        color: ${COLORS.FIELD_LABEL_TOOLTIP_TEXT};
        margin-right: 0.4rem;
        font-size: 1.6rem;
        line-height: 1rem;
      }
    `}
`
