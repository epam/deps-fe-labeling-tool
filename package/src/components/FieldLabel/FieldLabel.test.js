import React from 'react'
import { shallow } from 'enzyme'
import { FieldLabel } from './FieldLabel'
import { StyledTooltip } from './FieldLabel.styles'

describe('Component: FieldLabel', () => {
  let defaultProps
  let wrapper

  beforeEach(() => {
    defaultProps = {
      active: false,
      clickable: true,
      onClick: jest.fn(),
      name: 'Label :',
      required: true,
      className: 'test-class'
    }

    wrapper = shallow(<FieldLabel {...defaultProps} />)
  })

  it('should render correct layout', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should pass correct properties to the component', () => {
    expect(wrapper.props().active).toBe(defaultProps.active)
    expect(wrapper.props().clickable).toBe(defaultProps.clickable)
    expect(wrapper.props().className).toBe(defaultProps.className)
  })

  it('should not render tooltip if field is not required', () => {
    defaultProps.required = false

    wrapper = shallow(<FieldLabel {...defaultProps} />)

    expect(wrapper.find(StyledTooltip).exists()).toBe(false)
  })
})
