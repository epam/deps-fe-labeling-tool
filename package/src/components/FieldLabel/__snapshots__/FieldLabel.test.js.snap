// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: FieldLabel should render correct layout 1`] = `
<FieldLabelstyles__Label
  active={false}
  className="test-class"
  clickable={true}
>
  <FieldLabelstyles__StyledTooltip
    placement="left"
    required={true}
    title="Required Field"
  >
    Label :
  </FieldLabelstyles__StyledTooltip>
</FieldLabelstyles__Label>
`;
